### PR TITLE
JSIEVE-108 Rely on ActionDiscard instead of a hack

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/DiscardAction.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import org.apache.jsieve.mail.Action;
+import org.apache.jsieve.mail.ActionDiscard;
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailAddress;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+
+public class DiscardAction extends FileIntoAction implements MailAction {
```
Comment (quynhn) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/DiscardAction.java]: Could we test some kinds of MailAction?

Comment (chibenwa) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/DiscardAction.java]: Sure.

I did not do it as jsieve integration in James is missing unit tests...

**Comment by chibenwa** [comment]
About build: to be green, it demands to wipe out .m2 of ci, and if merge will delay release... Lets wait a bit

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/jsieve/DiscardActionTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        FakeMail mail = FakeMail.builder()
+            .recipients(MailAddressFixture.ANY_AT_JAMES, MailAddressFixture.OTHER_AT_JAMES)
+            .build();
+        ActionContext actionContext = mock(ActionContext.class);
+        when(actionContext.getRecipient()).thenReturn(MailAddressFixture.ANY_AT_JAMES);
+
+        DiscardAction.removeRecipient(mail, actionContext);
+
+        assertThat(mail.getRecipients()).containsOnly(MailAddressFixture.OTHER_AT_JAMES);
+    }
```
Comment (strokyl) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/jsieve/DiscardActionTest.java]: What if something bad happen and we have twice the recipient to remove

Comment (chibenwa) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/jsieve/DiscardActionTest.java]: You have a test showing that removing a non existing recipient has no effect ;-)

Comment (strokyl) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/jsieve/DiscardActionTest.java]: Yes but what I ask is different if you twice the recipient to remove in the mail will you remove only one instance of it or all of them ?

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**